validphys2/src/validphys/loader.py

```diff
@@ -370,19 +370,20 @@ def check_fkyaml(self, name, theoryID, cfac):
         metadata, fklist = pineparser.get_yaml_information(fkpath, theory.path)
         op = metadata["operation"]
 
-        # TODO:
-        #      at the moment there are no pineappl specific c-factors
-        #      so they need to be loaded from the NNPDF names / compounds files
         cfac_name = metadata["target_dataset"]
         # check whether there is a compound file
         cpath = theory.path / "compound" / f"FK_{cfac_name}-COMPOUND.dat"
         if cpath.exists():
-            # Get the target filenames
+            # Get the target filenames for old theories
             tnames = [i[3:-4] for i in cpath.read_text().split() if i.endswith(".dat")]
             cfactors = [self.check_cfactor(theoryID, i, cfac) for i in tnames]
         else:
-            cfactors = [self.check_cfactor(theoryID, cfac_name, cfac)]
-        ###
+            # load cf according to ymldb for new theories
+            cfac_name = metadata["operands"]
+            cfactors = []
+            for operand in cfac_name:
+                for fk_tab in operand:
+                    cfactors.append(self.check_cfactor(theoryID, fk_tab, cfac))
```

Comment:
Just noticed this won't work ootb for fktables with concatenations (you probably are not using any because it only happens for some of the LHC ones) so I will modify a bit the logic as well.

From your point of view it should make no difference, but wanted to give you a heads up in case I break something.

Reply:
okay thanks